### PR TITLE
Fix/2580/make sure all command line tools understand gz files

### LINKED ISSUE #2580: Make sure all command line tools understand .gz files
# Bug

## Expected Behavior

GIVEN a folder of cc.json.gz files

WHEN trying to merge them (and probably more commands)

THEN all files are combined

## Actual Behavior

The .gz files are not possible to be read by the merge filter.

## Steps to Reproduce the Problem

Use `ccsh merge`

## Specifications
- is released in [online-demo](https://maibornwolff.github.io/codecharta/visualization/app/index.html?file=codecharta.cc.json&file=codecharta_analysis.cc.json): yes 
- CodeCharta Version: 1.85.0
- OS: all
- Browser: not applicable

### PR DESCRIPTION
**Pull request by phanlezz**
# Make sure all command line tools understand .gz files

_**Please read the [CONTRIBUTING.md](https://github.com/MaibornWolff/codecharta/blob/main/CONTRIBUTING.md) before opening a PR.**_

Issue: #2580 

## Description

Enables .gz support for all analysis cli tools

Closes #2580

### CODE REVIEW

File: analysis/model/src/main/kotlin/de/maibornwolff/codecharta/serialization/CompressedStreamHandler.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    fun handleInput(input: InputStream): InputStream {
+        var content = input
+        if (!input.markSupported()) content = BufferedInputStream(input)
+        content.mark(2)
+        val byteArray = content.readNBytes(2)
+        content.reset()
+        if (isGzipByteHeader(byteArray)) return GZIPInputStream(content)
+        return content
+    }
```
Comment (ce-bo) [inline analysis/model/src/main/kotlin/de/maibornwolff/codecharta/serialization/CompressedStreamHandler.kt:17]: - Maybe find a better name of this function that expresses what the function is actually doing.
- Please always use `{}` for control / if / loop statements (even for one line if-blocks).

**Review by ce-bo — COMMENTED** [review]
It is looking good to me. It would be nice if anybody else could test this locally.

Another thing: We are able to "pipe in" a stream when executing a ccsh command (e.g. https://github.com/MaibornWolff/codecharta/blob/main/analysis/filter/StructureModifier/README.md). Would you please check, if it is possible to pipe in a gziped file?

Thank you.

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/)
 ### Added 🚀
 
 -   Added missing documentation for ccsh filters and csv exporter [#2831](https://github.com/MaibornWolff/codecharta/pull/2831)
+-   Added support for compressed files to all analysis filters and to `check` validation
```
Comment (IhsenBouallegue) [inline CHANGELOG.md:13]: Please add the PR link

**Comment by IhsenBouallegue** [comment]
We are trying to get away from using Spek for testing. For new tests, please use the JUnit library. If refactoring an old test that uses Spek please convert it to JUnit. It should be a simple syntax update.

File: CHANGELOG.md (hunk 2 above)
Comment (phanlezz) [reply CHANGELOG.md]: Done ✅ Good catch ty

File: analysis/model/src/main/kotlin/de/maibornwolff/codecharta/serialization/CompressedStreamHandler.kt (hunk 1 above)
Comment (phanlezz) [reply analysis/model/src/main/kotlin/de/maibornwolff/codecharta/serialization/CompressedStreamHandler.kt]: `handleInput` => `wrapInput`

**Comment by ce-bo** [comment]
> > It is looking good to me. It would be nice if anybody else could test this locally.
> > Another thing: We are able to "pipe in" a stream when executing a ccsh command (e.g. https://github.com/MaibornWolff/codecharta/blob/main/analysis/filter/StructureModifier/README.md). Would you please check, if it is possible to pipe in a gziped file?
> > Thank you.
> 
> It is possible to pipe in compressed files ✅ I just tested it.
> 
> Bildschirmfoto 2022-06-07 um 09 57 49

Thank you.
Does it work for other parsers/filters/tools as well?

**Review by IhsenBouallegue — APPROVED** [review]
LGTM 🥳

**Comment by phanlezz** [comment]
> [...]
> Thank you. Does it work for other parsers/filters/tools as well?

Works with:
- [ ]  Modify
- [ ]  RawTextParser


The other importers, exporters, and tools like merge, edgefilter, and check do not support to pipe into, or don't handle `cc.json` files

**Comment by ce-bo** [comment]
> > [...]
> > Thank you. Does it work for other parsers/filters/tools as well?
> 
> Works with:
> 
> * [x]   Modify
> * [x]   RawTextParser
> 
> The other importers, exporters, and tools like merge, edgefilter, and check do not support to pipe into, or don't handle `cc.json` files
> 
> @ce-bo Or was the implication behind the question and part of the issue, that **all other** importers/exporters should also support compressed files, e.g. `.xml.gz` ? Then I would need to change some more stuff 😄

No, I just wanted to be sure that to pipe in a file is still possible. Thank you :)

**Pull request merged**